common/core/nesteddisplay.h

```diff
@@ -17,20 +17,103 @@
 #ifndef COMMON_CORE_NESTEDDISPLAY_H_
 #define COMMON_CORE_NESTEDDISPLAY_H_
 
+#include <nativedisplay.h>
 #include <stdlib.h>
 #include <stdint.h>
 
+#include <linux/hyper_dmabuf.h>
 #include <memory>
-
+#include <map>
+#include "drmbuffer.h"
+#include <utils/threads.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+
+#include "hwcthread.h"
+#include "hwctrace.h"
 #include <nativedisplay.h>
+#define SURFACE_NAME_LENGTH    64
 
 namespace hwcomposer {
 
+struct HwcLayer;
+class NativeBufferHandler;
 class NestedDisplayManager;
 
+struct vm_header {
```

Comment:
Can these be just in cpp file than here ?

Reply:
no, these struct are member of  class nestedDisplay, I need place them in header file

Reply:
are they needed in header ?

Reply:
yes, you can see that I add a filed in NestedDisplay class(std::map<HWCNativeHandle, vm_buffer_info> mHyperDmaExportedBuffers;   // Track the  and hyper dmabuf metadata info mapping), so I can't move such define to cpp file